### ISSUE TITLE
S3: High connectivity check rate

### ISSUE THREAD

**Issue by casKd-dev** [nextcloud/server#16623]
### Steps to reproduce
1. Upload a file
2. List the file several times
3. Get S3 Bucket log
4. Count amount of HEAD requests to /

### Expected behaviour
These connectivity checks would be cached and reused

### Actual behaviour
The connectivity checks are done once every transaction

## Server configuration detail

**Operating system:** Linux 4.15.0-52-generic #56-Ubuntu SMP Tue Jun 4 22:49:08 UTC 2019 x86_64

**Webserver:** nginx/1.16.0 (fpm-fcgi)

**Database:** pgsql PostgreSQL 11.4 (Debian 11.4-1.pgdg90+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 6.3.0-18+deb9u1) 6.3.0 20170516, 64-bit

**PHP version:** 

7.3.6
Modules loaded: Core, date, libxml, pcre, zlib, filter, hash, readline, Reflection, SPL, session, cgi-fcgi, bz2, ctype, curl, dom, fileinfo, ftp, gd, gmp, iconv, intl, json, ldap, mbstring, openssl, pcntl, PDO, pgsql, posix, standard, SimpleXML, smbclient, sodium, sqlite3, xml, xmlwriter, zip, exif, imap, mysqlnd, pdo_pgsql, pdo_sqlite, Phar, xmlreader, pdo_mysql, apcu, igbinary, redis, memcached, imagick, mcrypt, libsmbclient, Zend OPcache

**Nextcloud version:** 16.0.3 - 16.0.3.0

**Updated from an older Nextcloud/ownCloud or fresh install:** Updated from older version

**Where did you install Nextcloud from:** Docker Hub (official image) (edited from linuxserver/nextcloud, still persists on official image)

<details><summary>Signing status</summary>

Array
(   [news] => Array       (           [EXTRA_FILE] => Array               (                   [vendor/arthurhoaro/favicon/resources/cache/url529662c1220c5fa892a9ddea95c0e3b1] => Array                       (                           [expected] =>                            [current] => 26d88555dc84368fd7717ecd3309f0289d02d22eab010e39c78efe757c219ea0dee57033cf7dace32fe67e178d99b593100bd43a43ff0db4992f3bf75a9e0296                       )                  [vendor/arthurhoaro/favicon/resources/cache/url8ffdefbdec956b595d257f0aaeefd623] => Array                       (                           [expected] =>                            [current] => 468c9ccd06fb8480c509e06f0fcd77102e7b4b4828cb00ffc59278987e121c5bcc53f0dce9de84041662625d737dd7f1cce5f514e5de6d5880c93ea7deff7c7b                       )                  [vendor/arthurhoaro/favicon/resources/cache/url9c29616a67ddd65de6607d389aeaecc9] => Array                       (                           [expected] =>                            [current] => 2811975115c31c7d8b83de056f6268ac734e6c446ba59831f4aaf412b17d02f5274f12cbd99c64d7af79e8b2d01b7932e7be9d1723c5debab4010eeae972f5b6                       )                  [vendor/arthurhoaro/favicon/resources/cache/urlc48a6859ceeaf822d91bf6e4002ccc7d] => Array                       (                           [expected] =>                            [current] => efbf559816aacc53edf653457b675d2eb15e79f08bc905d1999a4a51108325c29b72fac1b533924ef8d416a3de0989ea7762cd66ebef456da5c1a37309793f0e                       )                  [vendor/arthurhoaro/favicon/resources/cache/url598fb05e33be5c575ea322968325d047] => Array                       (                           [expected] =>                            [current] => babe32d9bcd3c9beff140ba3c160cf29590ec2bb7263ff6e94a7fb6e281ee5a4302023be96e4b261642ee79f1efb18e5630f02c20a0dca6573ee73de5aea1dce                       )                  [vendor/arthurhoaro/favicon/resources/cache/url01bb31b0cb2639e7fe73c1ee5b2a7696] => Array                       (                           [expected] =>                            [current] => 2701c8909e16470623d25f15fa0fdc4c2eb5abfd37b800d317734c89dba3992214a4f3be73969087fc094b3896c9f328ebb5094b89805e3bcc46b329849f71c7                       )                  [vendor/arthurhoaro/favicon/resources/cache/urlf921add8f140c52efa113f286d3651e0] => Array                       (                           [expected] =>                            [current] => c6820dd23c497569369714206dc6a87c529d59aee05ab199cf4374a40ee2b2dc4d3b5bdea17cfc78a553a1dc924ed21cd42db5c26519f6d4a32373b8e2b867f2                       )                  [vendor/arthurhoaro/favicon/resources/cache/urlf60c9ea2540638ac984ce3b13f5f303f] => Array                       (                           [expected] =>                            [current] => a0eba8923e060268df336cc85aac759ed81bc48f6d368f14e23f73a71921d3f919be7c059b1f35a34802d6c8ac94ec07e4338b4889f03cebd604b47b651b2eaa                       )                  [vendor/arthurhoaro/favicon/resources/cache/urlbc6fbc30fa255ab02d20802be0d8f6bb] => Array                       (                           [expected] =>                            [current] => 8d5ae7681080d45b13c367bbcbacca77ab96deb4e4be14a9f17600ac142b6376aeba9b46434c540424a197ac3451ad15f3b880fd417c8795cfdd6f0e05d89e0c                       )                  [vendor/arthurhoaro/favicon/resources/cache/urlb430ef0623520814170e46436903aed7] => Array                       (                           [expected] =>                            [current] => bd2f06c0b2b36fc88cc32d77c98b8df12cbad15947b4c70ef06efd3ba3fae357665a86169be6e0fc7663881c851194c9ae9b9569db5b46908a1c8ab4648d38c9                       )                  [vendor/arthurhoaro/favicon/resources/cache/url68c16ba7b54b82eb70361de4cb20d5c5] => Array                       (                           [expected] =>                            [current] => 48aad6080dd5b24693cd023c1565534ba08a621dd07059fc542549b61335e72fb712d62ed7acd70ea1df5c00caa28d46fe803e6082773ec18597949fc4ef7468                       )                  [vendor/arthurhoaro/favicon/resources/cache/urle1fed756f00dc5b373d93b0335608508] => Array                       (                           [expected] =>                            [current] => 1895b41a3797f90be4195305beb60405655ab1de4c77f6fd82b8cf1b384325dcc8cb344ee89c48edeb128b496a28ae5c27346b803256d2a716d3559c1193794b                       )                  [vendor/arthurhoaro/favicon/resources/cache/urld245406cb6c9f36be9064c92c34e12e1] => Array                       (                           [expected] =>                            [current] => db19d1de3fdf1bf9ad3619bc42010e761e2e9fe5aeac33720338e36d78da0187b48797dc89e5cce1f30e6179718cb889777cb4209ee6dc2887ec0ed607897541                       )                  [vendor/arthurhoaro/favicon/resources/cache/urle506ea4dc9486d471057e45c3ff3adf4] => Array                       (                           [expected] =>                            [current] => c86ddc442617a3dd2d71fbdab8803e018296d014a461b498894fd7432b1c2429ce4e45c58743b03a864df3baa1a8f39d77465e285a073d5f8daf7367a3243449                       )                  [vendor/arthurhoaro/favicon/resources/cache/urld6d429a24bb85d1ef122f6c34e32c2e1] => Array                       (                           [expected] =>                            [current] => b3982943bfa4694d424693196ffa30f354238f7544a75743c029ac028582f83d9ffe2bfbb938df861be52b6428cc8b8f0b723e9004fa0e877eb698544a494623                       )                  [vendor/arthurhoaro/favicon/resources/cache/url1e647548bdc288f2c9d3d0128253d740] => Array                       (                           [expected] =>                            [current] => ae1a747b83af15085b75181c6fdd9b27841e887d8b6c82776f8d9604e00ce9df27da1242ea2ea012f34ba6dcc3291fc2166935bbe6df02f2efacf3ebcdd182b1                       )                  [vendor/arthurhoaro/favicon/resources/cache/url86f59539dd529bf08baac69ebfca0e7b] => Array                       (                           [expected] =>                            [current] => 6a0c7e15247e402ba8e416514eb235703aada16a0c97ffce0761e3355aa693cbfb53171ff18064de355cb7f03fa1c0dc6b6c9d330b11a601ecb0d15980f04647                       )                  [vendor/arthurhoaro/favicon/resources/cache/url0e5282c1ec3f659dda0c80c74c543287] => Array                       (                           [expected] =>                            [current] => df893fda8e1c339ebb085402ec1c12b96c45c67c2e8ebc412393cddfd4f1c4f404e454163b932809233ef645023b6e1e62aee33232cdb5c23424b077bd38bd1b                       )                  [vendor/arthurhoaro/favicon/resources/cache/url54cad62867fea7c49a83b71d1d8d7c38] => Array                       (                           [expected] =>                            [current] => cbd2b4529e7673e26ba8d8f147be211b7f90c60e3acfb8e9704cf6f6b8073a5302c5b08e13003c7c35559db0ddd4e99dd5f8f30c13c37b286619cd7e99703f6b                       )                  [vendor/arthurhoaro/favicon/resources/cache/urla8bc9ad6485d9c7d084a3393ae1e9c89] => Array                       (                           [expected] =>                            [current] => 3c2ac8539f1d1700cc4368cf81b64c45e79c82ab043529cbf5e741bfeea679b353345fdae58e8e52b19a54465ab84def8d564041b4bb99b2cff6bae9d660d356                       )                  [vendor/arthurhoaro/favicon/resources/cache/url30dd8fd5626b7e36783516a6e238baf2] => Array                       (                           [expected] =>                            [current] => 6ff21c29e41497ef890524ee767f5b75aa99fd66a23c1e5442c3b16944bc2af4c1711ef28dc1e22ab2a701bc255e4f6580b1fa2016bde22fd9883882168b5f99                       )                  [vendor/arthurhoaro/favicon/resources/cache/urlf841e611ccc9945a054d4dd416c917c4] => Array                       (                           [expected] =>                            [current] => 0d35d6095b3e1935ec652b14da1f6b969a059cc4d9809ddf5cc9ad23cc409e8625c8ce8f32538546f2ed6ea0980235c9d8e97070f94011cc6de5e3c439aecd6a                       )                  [vendor/arthurhoaro/favicon/resources/cache/url722aad11d08f74a15707b7a8d137a1df] => Array                       (                           [expected] =>                            [current] => f6f5fbf403a16ddf0290d1f021ff8ed9f7b8ff6770d12e86ff6e724315e0a1ded401aa17256d3ac785711ed3a4aef9e944cd1303e494af8a461af6f0b28c3019                       )                  [vendor/arthurhoaro/favicon/resources/cache/urlbf1dde573f6f6d57d901a18aab808524] => Array                       (                           [expected] =>                            [current] => 21b97f7c5ecd3a62aac459ccd9448127e4413aa1b837a31e3088750a13d433fa67f957e9ab43cd0110d2f9ef656c52b3e2afa2f7538fc9baa88cee1dd96e82e8                       )                  [vendor/arthurhoaro/favicon/resources/cache/url2438f8d94c3cfbd27134e96e240958b9] => Array                       (                           [expected] =>                            [current] => 1c09b8a7137f2ad0bd5c600386f05078283641551b73e6192a110c40ab1c03bb81ffe46787e12da561df605d9e8751da5c7be502195286d6fc5ffb0ba916e750                       )                  [vendor/arthurhoaro/favicon/resources/cache/url48e8795e5472d4bb89d1cce5a9fc485d] => Array                       (                           [expected] =>                            [current] => 09f875683f013732f5a146c11a819664a003503912c0bd78b07fc170ae9d80926cab859ce182dbdd3c510de7edbf6f05cf5a6c56085a3803902fe25490b163b8                       )                  [vendor/arthurhoaro/favicon/resources/cache/urlc45b45240388b61ee7f3e1176507ca38] => Array                       (                           [expected] =>                            [current] => e45d9a8d73aa09ab0185056dd3eba4b76ec0aaef81f76b6b550de3176b3388f8099f5015a954710b85f2a1f6dcefba37a0dfd540a04092efe0a8115c5b465a00                       )                  [vendor/arthurhoaro/favicon/resources/cache/url49e42c86929168e1a00c9af1a4669f96] => Array                       (                           [expected] =>                            [current] => 9016f87cc1c0239ef4635f93b51c773898fe3a32bc3e39003bf589aeff213502540d313fee1d8fece96f2a28cc20122c7047db40d929191fcc93f2aee2db7189                       )                  [vendor/arthurhoaro/favicon/resources/cache/url6dd7716311d08856ff3e3936b8809312] => Array                       (                           [expected] =>                            [current] => a4a9e6d52a4dc011e1ada54e8ed2b536e36df83508a6e720cf10ab3e6565e6138a173e1058a3c8eb8a018f7c35f94c97c58516fda4a1633d6ba030e8903be482                       )                  [vendor/arthurhoaro/favicon/resources/cache/url345abb0a1ae87c15ec8c905882a3f0d0] => Array                       (                           [expected] =>                            [current] => 55e5d801d5404de0588a62ca2a28c39a0219a7cbdef6f912bcb71b335e84db08fa5275c74e499e979bbb7678dcdf888e813b150713c8130fe31fba6b2e329058                       )                  [vendor/arthurhoaro/favicon/resources/cache/url03f2eb192df2ddda43946eb98e7002f4] => Array                       (                           [expected] =>                            [current] => d7e3fa89d58ad0102064b81549cb965f1cca545777651121368d8080633e27c75ed8ddb57c9c047b531c8b3cd246be6d1d7c5b90abf3d313efd26d3d570cd8cb                       )                  [vendor/arthurhoaro/favicon/resources/cache/url4045ddfd2ca7819170a4da97c8b7b5ea] => Array                       (                           [expected] =>                            [current] => 9d057cff84af2a79024417798a2c9c832f84bc72ca128e306e0e598f059349d5365968f5c74bd8ed7fb542be1da22498c0b8dc4b8640343414d89a59bafccadc                       )                  [vendor/arthurhoaro/favicon/resources/cache/url18b1c7dc72deacdc899a36059d9d73fc] => Array                       (                           [expected] =>                            [current] => 31ea3fc34326bab6ccd9343d2fe76da4e4fd8360bd4dea43b9f1333811f1ccfccf5b034792fe17e7df113c4d75b5caaffd774c137675660c39102b9fc5c6258b                       )                  [vendor/arthurhoaro/favicon/resources/cache/url3097fca9b1ec8942c4305e550ef1b50a] => Array                       (                           [expected] =>                            [current] => 07cb5698227798458dd8855a4f39c8ad9c770105cba9e4538a1babf08de083fe747252e9806b142e7c6b5f167b48a5cd58e909aa03aa5a71a99de0769d11b177                       )                  [vendor/arthurhoaro/favicon/resources/cache/urldf42d281ac2498a4f9aa55f82ec3332b] => Array                       (                           [expected] =>                            [current] => fb80b161128ee6ebb9420e51d64e3f4532b0cc6c44858100b7f85a569d0cf5239c68d55a833735d513c48e600a9d805823b5dd179e81d371f2363abb7e25e8a5                       )              )      )

)

</details>

<details><summary>List of activated apps</summary>

```
Enabled:- accessibility: 1.2.0- activity: 2.9.1- audioplayer: 2.7.2- bookmarks: 1.0.6- bruteforcesettings: 1.4.0- calendar: 1.7.0- cloud_federation_api: 0.2.0- comments: 1.6.0- dav: 1.9.2- deck: 0.6.4- federatedfilesharing: 1.6.0- federation: 1.6.0- files: 1.11.0- files_external: 1.7.0- files_markdown: 2.0.6- files_pdfviewer: 1.5.0- files_rightclick: 0.13.0- files_sharing: 1.8.0- files_texteditor: 2.8.0- files_trashbin: 1.6.0- files_versions: 1.9.0- files_videoplayer: 1.5.0- firstrunwizard: 2.5.0- groupfolders: 4.0.3- issuetemplate: 0.5.0- logreader: 2.1.0- lookup_server_connector: 1.4.0- mail: 0.15.1- metadata: 0.9.0- news: 13.1.6- nextcloud_announcements: 1.5.0- notifications: 2.4.1- oauth2: 1.4.2- onlyoffice: 2.4.0- password_policy: 1.6.0- privacy: 1.0.0- provisioning_api: 1.6.0- recommendations: 0.4.0- registration: 0.4.6- serverinfo: 1.6.0- sharebymail: 1.6.0- spreed: 6.0.3- support: 1.0.0- survey_client: 1.4.0- systemtags: 1.6.0- theming: 1.7.0- twofactor_backupcodes: 1.5.0- twofactor_totp: 2.1.2- updatenotification: 1.6.0- viewer: 1.0.0- workflowengine: 1.6.0
Disabled:- admin_audit- apporder- contacts- encryption- end_to_end_encryption- flowupload- gallery- keeweb- notes- ocdownloader- polls- richdocuments- twofactor_u2f- twofactor_yubikey- user_ldap

```
</details>

<details><summary>Configuration (config/config.php)</summary>

```
{   "memcache.local": "\\OC\\Memcache\\APCu",   "memcache.distributed": "\\OC\\Memcache\\Redis",   "memcache.locking": "\\OC\\Memcache\\Redis",   "datadirectory": "***REMOVED SENSITIVE VALUE***",   "version": "16.0.3.0",   "passwordsalt": "***REMOVED SENSITIVE VALUE***",   "secret": "***REMOVED SENSITIVE VALUE***",   "apps_paths": [       {           "path": "\/config\/www\/nextcloud\/apps",           "url": "\/apps",           "writable": false       },       {           "path": "\/config\/www\/nextcloud\/custom_apps",           "url": "\/custom_apps",           "writable": true       }   ],   "redis": {       "host": "***REMOVED SENSITIVE VALUE***",       "port": 6379   },   "instanceid": "***REMOVED SENSITIVE VALUE***",   "mail_from_address": "***REMOVED SENSITIVE VALUE***",   "trusted_domains": [       "***REMOVED SENSITIVE VALUE***"   ],   "dbtype": "pgsql",   "dbname": "***REMOVED SENSITIVE VALUE***",   "dbhost": "***REMOVED SENSITIVE VALUE***",   "dbport": "",   "dbtableprefix": "oc_",   "dbuser": "***REMOVED SENSITIVE VALUE***",   "dbpassword": "***REMOVED SENSITIVE VALUE***",   "overwrite.cli.url": "https:\/\/cloud.redxen.eu",   "overwriteprotocol": "https",   "trusted_proxies": "***REMOVED SENSITIVE VALUE***",   "installed": true,   "trashbin_retention_obligation": "auto,7",   "enable_previews": false,   "forwarded_for_headers": [       "HTTP_X_FORWARDED_FOR"   ],   "objectstore": {       "class": "OC\\Files\\ObjectStore\\S3",       "arguments": {           "bucket": "rxnc",           "key": "***REMOVED SENSITIVE VALUE***",           "secret": "***REMOVED SENSITIVE VALUE***",           "port": 443,           "use_ssl": true,           "region": "eu-central-1"       }   },   "filesystem_check_changes": 0,   "skeletondirectory": "",   "mail_smtpmode": "smtp",   "mail_sendmailmode": "smtp",   "mail_domain": "***REMOVED SENSITIVE VALUE***",   "mail_smtpauthtype": "LOGIN",   "mail_smtpsecure": "tls",   "mail_smtpauth": 1,   "mail_smtphost": "***REMOVED SENSITIVE VALUE***",   "mail_smtpport": "587",   "mail_smtpname": "***REMOVED SENSITIVE VALUE***",   "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",   "twofactor_enforced": false,   "config_is_read_only": true,   "htaccess.RewriteBase": "\/",   "twofactor_enforced_groups": [       "admin"   ],   "twofactor_enforced_excluded_groups": [],   "theme": "",   "loglevel": 2
}
```
</details>

**Are you using external storage, if yes which one:** S3

**Are you using encryption:** false

**Are you using an external user-backend, if yes which one:**

## Client configuration

**Browser:** Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Falkon/3.1.0 Chrome/73.0.3683.105 Safari/537.36

**Operating system:** Linux RXHome 5.2.4-zen1-1-zen #1 ZEN SMP PREEMPT Sun Jul 28 10:52:59 UTC 2019 x86_64 GNU/Linux

## Logs

<details><summary>S3 Snippet</summary>
REST.HEAD.BUCKET - "HEAD / HTTP/1.1" 200 - - - 14 13 "-" "aws-sdk-php/3.62.10 GuzzleHttp/6.3.3 curl/7.65.1 PHP/7.3.6" - jvhUHRNP2kS1fT7hAOw7BD3Uo4fEI2GFOmizgDZLTMnClxKFxsvpPEcCkczk8Tbta2vGMj75Bxw= SigV4 ECDHE-RSA-AES128-GCM-SHA256 AuthHeader rxnc.s3.eu-central-1.amazonaws.com -
REST.GET.OBJECT urn%253Aoid%253A18610 "GET /urn%3Aoid%3A18610 HTTP/1.0" 200 - 452232 452232 14 13 "-" "aws-sdk-php/3.62.10" - JN4vhq/fxdKhRIVDV5lqdfsHuEPdSJkRoM2Cn4I+0wEm2wqOc9dQjBOT3PiuSGuf+k9+7RpaP5k= SigV4 ECDHE-RSA-AES128-GCM-SHA256 AuthHeader rxnc.s3.eu-central-1.amazonaws.com -
REST.HEAD.BUCKET - "HEAD / HTTP/1.1" 200 - - - 13 12 "-" "aws-sdk-php/3.62.10 GuzzleHttp/6.3.3 curl/7.65.1 PHP/7.3.6" - HTVTiYm89K93hK47xepazKX4cxlwFQml+Taas9xekTb+FWZU5URjne9chLbinSUVf8BhrfDF5tc= SigV4 ECDHE-RSA-AES128-GCM-SHA256 AuthHeader rxnc.s3.eu-central-1.amazonaws.com -
REST.GET.OBJECT urn%253Aoid%253A18610 "GET /urn%3Aoid%3A18610 HTTP/1.0" 200 - 452232 452232 70 68 "-" "aws-sdk-php/3.62.10" - DNE3WG1DyOS9EkNE7TxLB9f4d8UBCR1cp6z+YVeB37tr4O3tjr3iRS7Qe6FaE01iX6jMxfWc5sc= SigV4 ECDHE-RSA-AES128-GCM-SHA256 AuthHeader rxnc.s3.eu-central-1.amazonaws.com TLSv1.2
REST.HEAD.BUCKET - "HEAD / HTTP/1.1" 200 - - - 1 1 "-" "aws-sdk-php/3.62.10 GuzzleHttp/6.3.3 curl/7.65.1 PHP/7.3.6" - QNONeEhmzY99gpsl+KUuF1NWwHvRofF4XIgw/FNjMA0ObEU90/3K1KTE63BVZDTwltHfo8lOsfM= SigV4 ECDHE-RSA-AES128-GCM-SHA256 AuthHeader rxnc.s3.eu-central-1.amazonaws.com TLSv1.2
REST.GET.OBJECT urn%253Aoid%253A18610 "GET /urn%3Aoid%3A18610 HTTP/1.0" 200 - 452232 452232 9 8 "-" "aws-sdk-php/3.62.10" - dey+MVD3CtzWKdwrzD4f3CGOnsSywAYo81L7XujD+n7Aiet8HT08s9xR5NPfVU6SWpeZH6HosLw= SigV4 ECDHE-RSA-AES128-GCM-SHA256 AuthHeader rxnc.s3.eu-central-1.amazonaws.com TLSv1.2
REST.HEAD.BUCKET - "HEAD / HTTP/1.1" 200 - - - 13 12 "-" "aws-sdk-php/3.62.10 GuzzleHttp/6.3.3 curl/7.65.1 PHP/7.3.6" - /9TTUBj8Th7ktOPHSc4e1Y90fRqWQlaDJwkFRmsIuCd5TH25CBfD5fuPlAATYui7RoISigQkJ34= SigV4 ECDHE-RSA-AES128-GCM-SHA256 AuthHeader rxnc.s3.eu-central-1.amazonaws.com TLSv1.2
REST.GET.OBJECT urn%253Aoid%253A18610 "GET /urn%3Aoid%3A18610 HTTP/1.0" 200 - 452232 452232 14 12 "-" "aws-sdk-php/3.62.10" - p5v7iq5yeKvAO1nHY7AGaSW+gu2Ama3z1PQLavuFRfdy36YQMlR4D/DzappxaZJpYpNcLAJ6Pq4= SigV4 ECDHE-RSA-AES128-GCM-SHA256 AuthHeader rxnc.s3.eu-central-1.amazonaws.com TLSv1.2
REST.HEAD.BUCKET - "HEAD / HTTP/1.1" 200 - - - 11 11 "-" "aws-sdk-php/3.62.10 GuzzleHttp/6.3.3 curl/7.65.1 PHP/7.3.6" - kxTLM3pW3Sr1rhHjFnVv0tYOOgiEHAyZ6916ycg7L/X4c7Hvz+ImIf3vetrDcSYTz69eT3298a4= SigV4 ECDHE-RSA-AES128-GCM-SHA256 AuthHeader rxnc.s3.eu-central-1.amazonaws.com TLSv1.2
REST.GET.OBJECT urn%253Aoid%253A18610 "GET /urn%3Aoid%3A18610 HTTP/1.0" 200 - 452232 452232 22 20 "-" "aws-sdk-php/3.62.10" - FMFKsuDiQUUuiEnmFRhiJDp3OpDLjkRAzQLbYF6g4zuC1dSgeoLD0lCMKTVilvSZhJZEAa9B7m4= SigV4 ECDHE-RSA-AES128-GCM-SHA256 AuthHeader rxnc.s3.eu-central-1.amazonaws.com TLSv1.2
REST.HEAD.BUCKET - "HEAD / HTTP/1.1" 200 - - - 12 11 "-" "aws-sdk-php/3.62.10 GuzzleHttp/6.3.3 curl/7.65.1 PHP/7.3.6" - JSzzQHXgaz+WnOmmIVhPk1D8+HHgU+lzZAmphAfRC7cF9KZGYy6z9NrAUgFGRGcpu6lUuG4dRxI= SigV4 ECDHE-RSA-AES128-GCM-SHA256 AuthHeader rxnc.s3.eu-central-1.amazonaws.com TLSv1.2
REST.GET.OBJECT urn%253Aoid%253A18610 "GET /urn%3Aoid%3A18610 HTTP/1.0" 200 - 452232 452232 39 37 "-" "aws-sdk-php/3.62.10" - eqmufJ/51zTwvlVoOCU2GgGjNdNfPuSUZyLinxCmikGaB9kTXuorc5cv/a2Zfj5I2KgfxLAnRsI= SigV4 ECDHE-RSA-AES128-GCM-SHA256 AuthHeader rxnc.s3.eu-central-1.amazonaws.com TLSv1.2
REST.HEAD.BUCKET - "HEAD / HTTP/1.1" 200 - - - 12 11 "-" "aws-sdk-php/3.62.10 GuzzleHttp/6.3.3 curl/7.65.1 PHP/7.3.6" - Aw0CLLgqqZ7QVQ+PripRGeKWJmztF6k04XgduCye2B7GEPmBt4y3bTc3jJrKKcIVmaWZTpFut5k= SigV4 ECDHE-RSA-AES128-GCM-SHA256 AuthHeader rxnc.s3.eu-central-1.amazonaws.com TLSv1.2
REST.GET.OBJECT urn%253Aoid%253A7376 "GET /urn%3Aoid%3A7376 HTTP/1.0" 200 - 148 148 70 70 "-" "aws-sdk-php/3.62.10" - CyKrBD2xZxEIKjl8LoamYH/OxyTXiBG3WYxAFPG4cqfjxBS3uzPkyG/ex/g+hLfEC9U29iz4BQo= SigV4 ECDHE-RSA-AES128-GCM-SHA256 AuthHeader rxnc.s3.eu-central-1.amazonaws.com TLSv1.2
</details>

The log shows that ~50% of the requests made by nextcloud are HEAD /

**Comment by kesselb** [comment]
The number of requests to s3 depends on [Filesystem check frequency](https://docs.nextcloud.com/server/16/admin_manual/configuration_files/external_storage_configuration_gui.html?highlight=once%20every%20direct#mount-options). Do you see less requests with "never"?

**Comment by casKd-dev** [comment]
> The number of requests to s3 depends on [Filesystem check frequency](https://docs.nextcloud.com/server/16/admin_manual/configuration_files/external_storage_configuration_gui.html?highlight=once%20every%20direct#mount-options). Do you see less requests with "never"?

Hello, this has happened with the mentioned configuration toggled on. I have tested with and without frequency checks and i have not seen any significant difference. You have also linked a External Storages page but i use S3 as Primary Storage.

**Comment by gcommit** [comment]
When i use nextcloud with S3 as primariy storage it is much slower than without S3 as primary storage.

I am using:
NC 16.0.3
Primary storage: S3 on Ceph

As i can see it takes:
login: 30-60 seconds
enter settings: 4 seconds
enter settings -> basic settings: 6 seconds
enter settings -> overview: 13 seconds
enter settings -> security: 10 seconds
enter apps: 6 seconds
back to files: 4 seconds

Could be better/faster ? :)

**Comment by casKd-dev** [comment]
@gcommit How is this even related with this issue?

**Comment by gcommit** [comment]
I was told there https://github.com/nextcloud/server/issues/7235#issuecomment-519832069
To have a look at this issue.

**Comment by RussianNeuroMancer** [comment]
@gcommit did you tried to follow @kesselb advice from message above?

**Comment by casKd-dev** [comment]
> @gcommit did you tried to follow @kesselb advice from message above?

> Hello, this has happened with frequency checks set to 'never' or 0 as listed in the config.php. I have tested with and without filesystem checks and i have not seen any significant difference.
> You have also linked a External Storages page but i use S3 as Primary Storage.

Yes, i did. However the checks still persist.

**Comment by gcommit** [comment]
> @gcommit did you tried to follow @kesselb advice from message above?

i did.

**Comment by acs-ferreira** [comment]
Those using S3 should try to proxy their requests an implement a cache.
Guide here for nginx: https://gist.github.com/mikhailov/9639593

Also read comments of the author bellow the code.

**Comment by kesselb** [comment]
https://github.com/nextcloud/server/pull/21728 could reduce the number of requests. Testers are always welcome ;)

**Comment by MorrisJobke** [comment]
> #21728 could reduce the number of requests. Testers are always welcome ;)

This will be shipped in Nextcloud 20. So feedback on this is very welcome as @kesselb said.

**Comment by IIPoliII** [comment]
Was it implemented in NC 20.0.1?
